optuna/samplers/_gp/sampler.py

```diff
@@ -60,12 +60,34 @@ class GPSampler(BaseSampler):
     This sampler fits a Gaussian process (GP) to the objective function and optimizes
     the acquisition function to suggest the next parameters.
 
-    The current implementation uses:
-        - Matern kernel with nu=2.5 (twice differentiable),
-        - Automatic relevance determination (ARD) for the length scale of each parameter,
-        - Gamma prior for inverse squared lengthscales, kernel scale, and noise variance,
-        - Log Expected Improvement (logEI) as the acquisition function, and
-        - Quasi-Monte Carlo (QMC) sampling to optimize the acquisition function.
+    The current implementation uses Matern kernel with nu=2.5 (twice differentiable) with automatic
+    relevance determination (ARD) for the length scale of each parameter.
+    The hyperparameters of the kernel are obtained by maximizing the marginal log-likelihood of the
+    hyperparameters given the past trials.
+    To prevent overfitting, Gamma prior is introduced for kernel scale and noise variance and
+    a hand-crafted prior is introduced for inverse squared lengthscales.
+
+    As an acquisition function, we use:
+        - log expected improvement (logEI) for single-objective optimization, and
+        - the summation of logEI and the logarithm of the feasible probability with the independent
+          assumption of each constraint for (black-box inequality) constrained optimization.
+
+    The optimization of the acquisition function is performed via:
+        1. Collect the best params from the past trials,
+        2. Collect ``n_preliminary_samples`` points using Quasi-Monte Carlo (QMC) sampling,
+        3. Choose the best point from the collected points,
+        4. Choose ``n_local_search`` points from the collected points using the roulette selection,
```

Comment:
There seems to be some ambiguity as to whether ``n_local_search`` points or ``n_local_search+1`` points are selected.

Reply:
I addressed your comment")